docs/proposals.rst

```diff
@@ -0,0 +1,112 @@
+.. proposals:
+
+Proposals
+=========
+
+This section should contain design and implementation proposals for Tarmak.
+Once the proposal has been implemented it should be moved to the regular user
+and/or developer documentation.
```

Comment:
Proposals generally should be preserved so as to serve as a concrete reference for why certain decisions were made. It also allows us to check if the implementation has drifted from the proposal.